connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResource.java

```diff
@@ -123,4 +171,15 @@ private String normalizedPluginName(String pluginName) {
             ? pluginName.substring(0, pluginName.length() - ALIAS_SUFFIX.length())
             : pluginName;
     }
+
+    String getAlias(String name) {
+        name = normalizedPluginName(name);
+        int lastIndexOf = name.lastIndexOf('.');
+        return lastIndexOf >= 0 ? name.substring(lastIndexOf + 1) : name;
+    }
+
+    private synchronized List<ConfigKeyInfo> doGetConfigDef(final String pluginName) {
+        PluginType pluginType = pluginsByType.getOrDefault(getAlias(pluginName), PluginType.UNKNOWN);
```

Comment:
Doesn't this introduce the possibility of conflict between two plugins (or I guess specifically connectors, since those are the only ones we strip suffixes from) which have different fully-qualified class names, but the same simple class name? Or where they would have the same simple class name, except that one ends with `Connector` and the other doesn't?

In practice this is unlikely to come up but if we support it at the moment, probably best to take care here to avoid introducing a potential regression, especially if someone for some reason wants to run, e.g., two different `MySqlSink` connectors on their worker.

Reply:
Also, it seems like using a default of `PluginType.UNKNOWN` here might be suboptimal. If someone wants to the view the config for a REST extension, for example, they'll end up seeing an error message later on (in `AbstractHerder::connectorPluginConfig`) that says something like "Invalid plugin type unknown. Valid types are..."

I think it'd be clearer to users if we could differentiate between these two cases:
1. User requests config for a plugin that does exist on the worker, but which we don't expose config information via the REST API for (such as a REST extension or a config provider)
2. User requests config for a plugin that doesn't exist on the worker

Status-wise, In the case of 1, a 400 response probably makes sense, but for 2, a 404 response might be more applicable.